presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -186,11 +188,18 @@ public Page getNextPage()
                         continue;
                     }
 
+                    String fieldName = fileSchema.getFields().get(fieldIndex).getName();
                     List<String> path = new ArrayList<>();
-                    path.add(fileSchema.getFields().get(fieldIndex).getName());
+                    path.add(fieldName);
                     if (ROW.equals(type.getTypeSignature().getBase())) {
```

Comment:
I think we should move this `if/else` block inside of `ParquetReader` and change its interface to have only a `readBlock (ColumnDescriptor, Type)` method.